Kernel/FileSystem/Ext2FS/Inode.cpp

```diff
@@ -428,24 +418,25 @@ InodeMetadata Ext2FSInode::metadata() const
     metadata.gid = inode_gid(m_raw_inode);
     metadata.link_count = m_raw_inode.i_links_count;
 
-    auto decode_time = [this](u32 const& time, u32 const& time_extra) {
-        // NOTE: All the *_extra fields have to be bounds-checked in case we have oddly-sized inodes.
-        // This is simply a correctness measure, since an OOB read wouldn't happen anyway due to the
-        // fact that we always store the raw inode as an ext2_inode_large.
-        if (is_within_inode_bounds(bit_cast<FlatPtr>(&m_raw_inode), bit_cast<FlatPtr>(&time_extra), sizeof(time_extra))) {
-            time_t seconds = decode_seconds_with_extra(time, time_extra);
-            u32 nanoseconds = decode_nanoseconds_from_extra(time_extra);
-            return UnixDateTime::from_unix_timespec({ .tv_sec = seconds, .tv_nsec = nanoseconds });
-        }
-
-        return UnixDateTime::from_seconds_since_epoch(static_cast<i32>(time));
+    auto decode_time_with_extra = [](u32 const& time, u32 const& time_extra) {
+        time_t seconds = decode_seconds_with_extra(time, time_extra);
+        u32 nanoseconds = decode_nanoseconds_from_extra(time_extra);
+        return UnixDateTime::from_unix_timespec({ .tv_sec = seconds, .tv_nsec = nanoseconds });
     };
 
-    metadata.atime = decode_time(m_raw_inode.i_atime, m_raw_inode.i_atime_extra);
-    metadata.mtime = decode_time(m_raw_inode.i_mtime, m_raw_inode.i_mtime_extra);
+    u64 usable_size = fs().usable_inode_size(m_raw_inode);
+    if (__builtin_offsetof(ext2_inode_large, i_atime_extra) + sizeof(m_raw_inode.i_atime_extra) <= usable_size)
```

Comment:
Anything that includes `<Kernel/Library/MiniStdLib.h>` should be able to use `offsetof` instead of having to use the "ugly" `__builtin_offsetof` builtin. (I don't know why we currently use `__builtin_offsetof` in so many places)